sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/Crashes.java

```diff
@@ -735,14 +739,17 @@ public void run() {
                         Map.Entry<UUID, ErrorLogReport> unprocessedEntry = unprocessedIterator.next();
                         ErrorLogReport errorLogReport = unprocessedEntry.getValue();
 
-                        /* TODO (getErrorAttachment): Re-enable error attachment when the feature becomes available. */
-//                        ErrorAttachment attachment = mCrashesListener.getErrorAttachment(errorLogReport.report);
-//                        if (attachment == null)
-//                            MobileCenterLog.debug(LOG_TAG, "CrashesListener.getErrorAttachment returned null, no additional information will be attached to log: " + errorLogReport.log.getId().toString());
-//                        else
-//                            errorLogReport.log.setErrorAttachment(attachment);
                         mChannel.enqueue(errorLogReport.log, ERROR_GROUP);
 
+                        Iterable<ErrorAttachmentLog> attachments = mCrashesListener.getErrorAttachment(errorLogReport.report);
+                        if (attachments == null) {
+                            MobileCenterLog.debug(LOG_TAG, "CrashesListener.getErrorAttachment returned null, no additional information will be attached to log: " + errorLogReport.log.getId().toString());
+                        }
+                        else {
+                            for (ErrorAttachmentLog attachment : attachments) {
+                                mChannel.enqueue(attachment, ERROR_GROUP);
```

Comment:
Since the static util method can return null, we can expect to have null references in the iterable.

And the static method could be bypassed, I really think we should move the validation call to this place (validation itself can be moved to another class if you want to minimize code file length increase).